pydy/tests/test_joints.py

```diff
@@ -0,0 +1,275 @@
+        assert self.pinjoint.parent_axis == self.parent.frame.x
+        assert self.pinjoint.child_axis == self.child.frame.y
+        point1 = self.parent.masscenter.locatenew(self.parent.frame, self.parent.frame.x)
+        point2 = self.child.masscenter.locatenew(self.child.frame, self.child.frame.y)
+        assert self.pinjoint.parent_point == point1
+        assert self.pinjoint.child_joint_piont == point2
+        assert self.pinjoint.parent_joint_vector == self.parent.frame.x
+        assert self.pinjoint.child_joint_vector == self.chidl.frame.y
+
+    def test_pinjoint_tuple_vector(self):
+        a, b, c, d, e, f = symbols('a b c d e f')
+        pinjoint = PinJoint('pinjoint', self.parent, self.child, par_point_vec_tuple=(a, b, c),
+                            child_point_vec_tuple=(d, e, f), parent_axis='x', child_axis='y')
+        assert pinjoint.parent_joint_vector == a * self.parent.frame.x + b * self.parent.frame.y + c * self.parent.frame.z
+        assert pinjoint.child_joint_vector == d * self.child.frame.x + e * self.child.frame.y + f * self.child.frame.z
+
+    def test_pinjoint_functions(self):
+
+        # part 1 assining parent-child relationship
+        self.pinjoint._set_parent_child_rel()
+        assert self.child.parent == self.parent
+        assert self.parent.child == self.child
+
+        # part 2 locating joint point in bodies
+        self.pinjoint._locate_joint_point()
+        child_masscenter = self.child.masscenter
+        parent_masscenter = self.parent.masscenter
+        parent_point = parent_masscenter.locatenew(self.name + '_Point', self.parent_joint_vector)
+        child_point = child_masscenter.locatenew(self.name + '_Point', self.child_joint_vector)
+        assert self.pinjoint.parent_joint_vector == parent_point
+        assert self.pinjoint.child_joint_vector == child_point
+
+        # part 3 aligning parent_axis and child_axis
+        self.join_frames()
+        # sample implementation
+        # self.child.frame.orient(self.parent.frame, 'Axis', [0, self.parent.frame.x])
+        assert cross(self.pinjoint.parent_axis, self.pinjoint.child_axis) == self.child.frame.z
+        assert cross(self.pinjoint.child_axis, self.pinjoint.parent_axis) == - self.parent.frame.z
+        self.align_axes()
+        parent_axis = self.pinjoint.parent_axis
+        child_axis = self.pinjoint.child_axis
+        # sample implementation
+        # perpendicular_axis_in_parent = - cross(parent_axis, child_axis)
+        # angle_between_axes = acos(dot(self.pinjoint.child_axis, self.pinjoint.parent_axis))
+        # self.child.frame.orient(self.parent.frame, 'Axis',
+        #                        [acos(dot(parent_axis, child_axis)/(parent_axis.magnitude() * child_axis.magnitude())),
+        #                         cross(parent_axis, child_axis)])
+        assert acos(dot(parent_axis, child_axis)/(parent_axis.magnitude() * child_axis.magnitude())) == 0
```

Comment:
This will probably need to be `assert trigsimp(expr) == 0`.

So here you make sure that the angle between the two vectors is zero after alignment. That's good.